runtimevar/awssecretsmanager/awssecretsmanager_test.go

```diff
@@ -81,60 +83,92 @@ func (h *harness) MakeWatcher(_ context.Context, name string, decoder *runtimeva
 
 func (h *harness) CreateVariable(ctx context.Context, name string, val []byte) error {
 	svc := secretsmanager.New(h.session)
-	awsName := aws.String(name)
-	token := aws.String(generateClientRequestToken(name, val))
-
-	// From AWS Secrets Manager docs:
-	// An asynchronous background process performs the actual secret deletion, so there
-	// can be a short delay before the operation completes. If you write code to
-	// delete and then immediately recreate a secret with the same name, ensure
-	// that your code includes appropriate back off and retry logic.
-	var backoff gax.Backoff
-	if *setup.Record {
-		backoff.Initial = 5 * time.Second
-		backoff.Max = 30 * time.Second
-	} else {
-		backoff.Max = time.Millisecond
-	}
-
-	return retry.Call(ctx, backoff,
-		func(err error) bool {
-			if awsErr, ok := err.(awserr.Error); ok {
-				return awsErr.Code() == secretsmanager.ErrCodeInvalidRequestException
-			}
-
-			return false
-		},
-		func() error {
-			_, err := svc.CreateSecretWithContext(ctx, &secretsmanager.CreateSecretInput{
-				Name:               awsName,
-				ClientRequestToken: token,
-				SecretBinary:       val,
-			})
-
-			return err
-		},
-	)
+	_, err := svc.CreateSecretWithContext(ctx, &secretsmanager.CreateSecretInput{
+		Name:               aws.String(name),
+		ClientRequestToken: aws.String(generateClientRequestToken(name, val)),
+		SecretBinary:       val,
+	})
+	if err != nil {
+		return err
+	}
+	// Secret Manager is only eventually consistent, so we insert a delay
+	// and verification that the mutation was applied. This is still not a guarantee
+	// but in practice seems to work well enough to make tests repeatable.
+	for {
+		if *setup.Record {
+			time.Sleep(consistencyDelay)
+		}
+		_, err := svc.GetSecretValueWithContext(ctx, &secretsmanager.GetSecretValueInput{
+			SecretId: aws.String(name),
+		})
+		if err == nil {
+			break
+		}
+	}
+	return nil
 }
 
-var errNotSupported = errors.New("method not supported")
-
-func (h *harness) UpdateVariable(_ context.Context, _ string, _ []byte) error {
-	return errNotSupported
+func (h *harness) UpdateVariable(ctx context.Context, name string, val []byte) error {
+	svc := secretsmanager.New(h.session)
+	_, err := svc.PutSecretValueWithContext(ctx, &secretsmanager.PutSecretValueInput{
+		ClientRequestToken: aws.String(generateClientRequestToken(name, val)),
+		SecretBinary:       val,
+		SecretId:           aws.String(name),
+	})
+	if err != nil {
+		return err
+	}
+	// Secret Manager is only eventually consistent, so we insert a delay
+	// and verification that the mutation was applied. This is still not a guarantee
+	// but in practice seems to work well enough to make tests repeatable.
+	for {
+		if *setup.Record {
+			time.Sleep(consistencyDelay)
+		}
+		getResp, err := svc.GetSecretValueWithContext(ctx, &secretsmanager.GetSecretValueInput{
+			SecretId: aws.String(name),
+		})
+		if err == nil && bytes.Equal(getResp.SecretBinary, val) {
```

Comment:
Same here: I think we should retry only on the `resource not found` error and non-equal secret value.

Reply:
Done.